### DISCUSSION TITLE
Move all integration tests to EF Core?

### DISCUSSION THREAD

**Discussion by jbogard** [LuckyPennySoftware/AutoMapper#3571] [Ideas]
Personally I haven't used EF6 for a few years, seems like targeting EF Core would be more valuable.

Instead of authoring two sets of integration tests, one for EF6 and one for EF Core, just target EF Core

@lbargaoanu thoughts?

**Comment by lbargaoanu** [comment]
As long as we have only one EF provider in this repo, I don't care which one is it :)

**Reply by jbogard** [reply]
Yeah that was my thought, instead of duplicating tests, just retarget to EF Core.

**Reply by jbogard** [reply]
I haven't used EF6 in maybe 4 years or so, and since I only maintain AutoMapper personally for my own paying clients and projects, it doesn't make sense for me to care about supporting features in our query expressions for a dependency none of my clients or projects have used for many years.

I would rather fix bugs to make EF Core work better than EF6. The kinda downside is there are even other query providers that I use that AutoMapper doesn't quite support (MongoDB and Cosmos DB are two of my recent examples). The integration tests are merely "here's the query provider we primarily target" and if the others work too, well great.

**Reply by mycroes** [reply]
I understand and I don't have any hard feelings about that, it would just be easier for me if EF6 was 'officially' supported as well. If you move all the tests to EF Core that doesn't mean anything will break for EF6, but in the long run changes might affect EF6 and go unnoticed because EF6 is no longer a test target. OTOH in the long run I don't want to be supporting EF6 applications either, so I'm not necessarily worried about this, I definitely prefer proper EF Core support over EF6 for future AutoMapper versions.

**Reply by lbargaoanu** [reply]
@mycroes If you're interested in moving the integration tests from EF6 to EF Core, that would be an welcome contribution. You can PR my branch.
